### PR TITLE
chore: implement operations

### PR DESCRIPTION
**Pull request by jrandolf-google**
This PR adds operations which are a new `PromiseLike` object (actually they _are_ `Promise`s) that allow declarative side effects through the `effect` method.

See tests for usage.

### CODE REVIEW

**Review by OrKoN — CHANGES_REQUESTED** [review]
Let's not land anything until we finalised the design.

File: packages/puppeteer-core/src/common/Operation.ts
Comment (OrKoN) [inline packages/puppeteer-core/src/common/Operation.ts]: why don't we throw in a sync way when effect is called? async errors won't be caught by this try/catch.

File: test/src/operation.spec.ts
Comment (OrKoN) [inline test/src/operation.spec.ts:18]: it should be possible to import from `puppeteer-core` now

File: packages/puppeteer-core/src/common/Operation.ts
Comment (OrKoN) [inline packages/puppeteer-core/src/common/Operation.ts]: I guess this needs to be removed before landing?

File: packages/puppeteer-core/src/common/Operation.ts
Comment (OrKoN) [inline packages/puppeteer-core/src/common/Operation.ts]: let's not land as public before we can finalise the entire feature.

**Review by OrKoN — APPROVED** [review]

File: test/src/operation.spec.ts
Comment (jrandolf) [reply test/src/operation.spec.ts:18]: The `create` method is an internal only method.

**Pull request merged**
